Runtime/Data/Collection/ObservableList.cs

```diff
@@ -43,38 +63,120 @@ public abstract class ObservableList<TElement, TEvent> : MonoBehaviour where TEv
         /// <summary>
         /// The collection to observe changes of.
         /// </summary>
-        public IReadOnlyList<TElement> Elements => elements;
+        public IReadOnlyList<TElement> ReadOnlyElements => Elements;
+
+        /// <summary>
+        /// Checks to see if the collection contains the given element.
+        /// </summary>
+        /// <param name="element">The element to search for.</param>
+        /// <returns>Whether the element is found.</returns>
+        [RequiresBehaviourState]
+        public virtual bool Contains(TElement element)
+        {
+            if (Elements.Contains(element))
+            {
+                Found?.Invoke(element);
+                return true;
+            }
+
+            NotFound?.Invoke(element);
+            return false;
+        }
+
+        /// <summary>
+        /// Checks to see if the collection contains the given element.
+        /// </summary>
+        /// <param name="element">The element to search for.</param>
+        [RequiresBehaviourState]
+        public virtual void DoContains(TElement element)
+        {
+            Contains(element);
+        }
 
         /// <summary>
-        /// Adds an element to the start of the collection.
+        /// Adds an element to the end of the collection.
         /// </summary>
         /// <param name="element">The element to add.</param>
         [RequiresBehaviourState]
-        public virtual void AddToStart(TElement element)
+        public virtual void Add(TElement element)
         {
-            elements.Insert(0, element);
+            Elements.Add(element);
             EmitAddEvents(element);
         }
 
         /// <summary>
-        /// Adds an element to the end of the collection.
+        /// Adds an element to the given index of the collection.
         /// </summary>
+        /// <remarks>
+        /// Allows the use of a wrapped and clamped index to prevent indices being out of bounds and doing negative queries such as `-1` sets the last element.
+        /// </remarks>
         /// <param name="element">The element to add.</param>
+        /// <param name="index">The index to add at.</param>
```

Comment:
```suggestion
        /// <param name="index">The index to add at. In case this index is out of bounds for the collection it will wrap around.</param>
```